### PR TITLE
T: add test that intentions have descriptions and templates

### PR DESCRIPTION
**Pull request by Kobzol**
This PR adds a test to all intentions that they have proper description and templates (inspired by a similar test in inspection test base class).

Blocked on https://github.com/intellij-rust/intellij-rust/pull/6050

### CODE REVIEW

File: src/test/kotlin/org/rust/ide/intentions/RsIntentionTestBase.kt [hunk 1]
```diff
@@ -11,9 +11,27 @@ import com.intellij.util.ui.UIUtil
 import org.intellij.lang.annotations.Language
 import org.rust.RsTestBase
 import org.rust.fileTreeFromText
+import java.nio.file.Paths
 
 abstract class RsIntentionTestBase(val intention: IntentionAction) : RsTestBase() {
 
+    fun testIntentionHasDocumentation() {
```
Comment (Undin) [inline src/test/kotlin/org/rust/ide/intentions/RsIntentionTestBase.kt:18]:
```suggestion
    fun `test intention has documentation`
```

Comment (Undin) [inline src/test/kotlin/org/rust/ide/intentions/RsIntentionTestBase.kt:18]: Let's check it only if `intention` is inheritor of `RsElementBaseIntentionAction`. We should check documentation only of our intentions. But tests can invoke any intention (see `org.rust.ide.intentions.InjectLanguageIntentionTest`)

File: src/test/kotlin/org/rust/ide/intentions/RsIntentionTestBase.kt [hunk 2]
```diff
@@ -11,9 +11,27 @@ import com.intellij.util.ui.UIUtil
...
+            ?: error("No intention description for ${intention.javaClass} ($description)")
+        checkHtmlStyle(text)
+
+        val afterTemplatePath = Paths.get(directory, "after.rs.template").toString()
+        getResourceAsString(afterTemplatePath)
+            ?: error("No after.rs.template found for ${intention.javaClass} ($afterTemplatePath)")
+
+        val beforeTemplatePath = Paths.get(directory, "before.rs.template").toString()
+        getResourceAsString(beforeTemplatePath)
+            ?: error("No before.rs.template found for ${intention.javaClass} ($beforeTemplatePath)")
```
Comment (Undin) [inline src/test/kotlin/org/rust/ide/intentions/RsIntentionTestBase.kt:32]: We use this similar code three times in a row. Maybe it's worth to extract it into separate method?

**Review by Undin — COMMENTED** [review]
Great!
Let's rebase onto master to fix test for `CreateFunctionIntention` intention

**Review by Undin — APPROVED** [review]
Thanks!
bors r+

**Comment by Undin** [comment]
bors retry

**Comment by Kobzol** [comment]
bors retry

**Comment by Kobzol** [comment]
@Undin is it possible to see why the merge failed? I can't see it on GitHub nor in BORS.

**Comment by Undin** [comment]
@Kobzol Yep.
image
image

**Comment by Kobzol** [comment]
Oh, I knew that, but somehow I missed the red cross marks :man_facepalming: Thanks :)

**Pull request merged**